test/envoye2e/http_metadata_exchange/http_metadata_exchange_test.go

```diff
@@ -1,167 +0,0 @@
-	"istio.io/proxy/test/envoye2e/env"
-)
-
-const metadataExchangeIstioDownstreamConfigFilter = `- name: envoy.filters.network.metadata_exchange
-  config:
-    protocol: istio2`
-
-const metadataExchangeIstioStatsServerFilter = `- name: envoy.filters.http.wasm
-  config:
-    config:
-      root_id: "stats_inbound"
-      vm_config:
-        runtime: envoy.wasm.runtime.null
-        code:
-          local: { inline_string: "envoy.wasm.stats" }
-      configuration: |
-        { "debug": "false", max_peer_cache_size: 20, field_separator: ";.;" }`
-const metadataExchangeIstioUpstreamConfigFilterChain = `filters:
-- name: envoy.filters.network.upstream.metadata_exchange
-  typed_config: 
-    "@type": type.googleapis.com/envoy.tcp.metadataexchange.config.MetadataExchange
-    protocol: istio2`
-
-const metadataExchangeIstioStatsClientFilter = `- name: envoy.filters.http.wasm
-  config:
-    config:
-      root_id: "stats_outbound"
-      vm_config:
-        runtime: envoy.wasm.runtime.null
-        code:
-          local: { inline_string: "envoy.wasm.stats" }
-      configuration: |
-        { "debug": "false", max_peer_cache_size: 20, field_separator: ";.;" }`
-
-const tlsContext = `tls_context:
-  common_tls_context:
-    alpn_protocols:
-    - istio2
-    tls_certificates:
-    - certificate_chain: { filename: "testdata/certs/cert-chain.pem" }
-      private_key: { filename: "testdata/certs/key.pem" }
-    validation_context:
-      trusted_ca: { filename: "testdata/certs/root-cert.pem" }
-  require_client_certificate: true`
-
-const clusterTLSContext = `tls_context:
-  common_tls_context:
-    alpn_protocols:
-    - istio2
-    tls_certificates:
-    - certificate_chain: { filename: "testdata/certs/cert-chain.pem" }
-      private_key: { filename: "testdata/certs/key.pem" }
-    validation_context:
-      trusted_ca: { filename: "testdata/certs/root-cert.pem" }`
-
-var (
-	statsConfig          = driver.LoadTestData("testdata/bootstrap/stats.yaml.tmpl")
-	outboundNodeMetadata = driver.LoadTestData("testdata/client_node_metadata.json.tmpl")
-	inboundNodeMetadata  = driver.LoadTestData("testdata/server_node_metadata.json.tmpl")
-)
-
-// Stats in Client Envoy proxy.
-var expectedClientStats = map[string]int{
-	"cluster.client.metadata_exchange.alpn_protocol_found":      1,
-	"cluster.client.metadata_exchange.alpn_protocol_not_found":  0,
-	"cluster.client.metadata_exchange.initial_header_not_found": 0,
-	"cluster.client.metadata_exchange.header_not_found":         0,
-	"cluster.client.metadata_exchange.metadata_added":           1,
-}
-
-// Stats in Server Envoy proxy.
-var expectedServerStats = map[string]int{
-	"metadata_exchange.alpn_protocol_found":      1,
-	"metadata_exchange.alpn_protocol_not_found":  0,
-	"metadata_exchange.initial_header_not_found": 0,
-	"metadata_exchange.header_not_found":         0,
-	"metadata_exchange.metadata_added":           1,
-}
-
-func TestHttpMetadataExchange(t *testing.T) {
```

Comment:
@gargnupur fyi, I removed this test, since it is not executed and we will need to rewrite it any way. Let me know if you are ok with this or not.

Reply:
Sure :) We can always come back to this PR if we decide to do metadata exchange using Alpn Filter...